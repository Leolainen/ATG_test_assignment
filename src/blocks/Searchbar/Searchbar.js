import Button from "components/Button";
import Input from "components/Input";
import PropTypes from "prop-types";
import React from "react";
import styles from "./styles.module.css";

const Searchbar = React.forwardRef(function Searchbar(props, ref) {
  const { className, onSearchClick, ...other } = props;

  return (
    <Input
      ref={ref}
      label="Search for a gametype:"
      type="search"
      cta={<Button onClick={onSearchClick}>Search</Button>}
      className={styles.root}
      placeholder="Search for V4, V64, V65 or V75 ..."
      {...other}
    />
  );
});

Searchbar.propTypes = {
  className: PropTypes.string,
};

Searchbar.uiName = "Searchbar";

export default Searchbar;
