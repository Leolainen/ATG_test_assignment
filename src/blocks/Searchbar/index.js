// export { default } from "./Searchbar.js";

import { fetchProductByType } from "api";
import { useAppContext } from "containers/App/AppContext";
import PropTypes from "prop-types";
import React, { useState } from "react";
import Searchbar from "./Searchbar";

const EnhancedSearchbar = React.forwardRef(function EnhancedSearchbar(
  props,
  ref
) {
  const { setSearchResults, onFetchStart, onFetchEnd } = useAppContext();
  const [searchValue, setSearchValue] = useState("");

  const handleChange = (event) => {
    setSearchValue(event.target.value);
  };

  const fetchGames = async () => {
    const results = await fetchProductByType(searchValue);
    setSearchResults(results);
    onFetchEnd();
  };

  const handleSearch = () => {
    onFetchStart();
    fetchGames();
  };

  return (
    <Searchbar
      ref={ref}
      onChange={handleChange}
      value={searchValue}
      onSearchClick={handleSearch}
    />
  );
});

EnhancedSearchbar.propTypes = {
  className: PropTypes.string,
};

EnhancedSearchbar.uiName = "EnhancedSearchbar";

export default EnhancedSearchbar;
