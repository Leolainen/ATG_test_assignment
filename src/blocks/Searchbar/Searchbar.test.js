import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Searchbar from "./Searchbar";

describe("<Searchbar />", () => {
  afterEach(cleanup);

  describe("events", () => {
    it("should fire onSearchClick callback when clicked", () => {
      const mockFn = jest.fn();
      const { getByTestId } = render(<Searchbar onSearchClick={mockFn} />);
      fireEvent.click(getByTestId("button"));

      expect(mockFn).toHaveBeenCalled();
    });
  });

  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<Searchbar />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
