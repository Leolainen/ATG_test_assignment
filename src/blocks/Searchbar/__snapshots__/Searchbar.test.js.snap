// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Searchbar /> Snapshots should render 1`] = `
<DocumentFragment>
  <div
    class="root root"
  >
    <label
      class="label"
    >
      Search for a gametype:
    </label>
    <div
      class="inner"
    >
      <input
        class="input"
        placeholder="Search for V4, V64, V65 or V75 ..."
        type="search"
      />
      <div
        class="cta"
      >
        <button
          class="root color"
          data-testid="button"
        >
          <span
            class="label"
          >
            Search
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
